design/protocol-changes/ledger.tex

```diff
@@ -5,13 +5,14 @@ \section{Changes to the Ledger}
 Peras requires new protocol parameters, see \cref{sec:protocol parameters}, which should be a routine change.
 
 Apart from that, the Ledger needs to be modified to account for Peras certificates which are included on chain to coordinate the end of a cooldown period.
+We note that certificate are morally part of the Consensus layer (in particular, validating them requires the epoch nonce, which no other ledger rule does); however, certificates are likely too large to be stored in a header.
 
 Concretely, we propose the following checks for a block body $B$ containing a certificate \cert{}:
 \begin{enumerate}
 \item
   The certificate \cert{} must be valid.
 \item
-  The round of \cert{} must be strictly greater than the round than the previous certificate on chain.
+  The round of \cert{} must be strictly greater than the round of the previous certificate on chain, and not larger than the round corresponding to the slot of $B$.
```

Comment:
In fact, it should be strictly smaller than the round corresponding to the slot of B, no?

Reply:
Added motivation for this rule:

> This ensures that $\texttt{cert}$ is neither too old nor too new, such that the ledger state that $B$ is validated against contains the necessary information to validate $r$.

Also realized that we need to add an explicit lower bound on the round number, so did that too.